src/android/AuthenticationDialog.java

```diff
@@ -0,0 +1,171 @@
+package org.apache.cordova.inappbrowser;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.res.Resources;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.view.inputmethod.EditorInfo;
+import android.widget.TextView;
+import android.widget.TextView.OnEditorActionListener;
+import android.graphics.Paint;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+
+/**
+ * Implements authentication dialog.
+ */
+public class AuthenticationDialog {
```

Comment:
@brianweyer Just a question: Why do not inherited from `Dialog`?

Reply:
I probably could have. This was a pretty quick implementation to get use the functionality we needed.